packages/vite/src/module-runner/runner.ts

```diff
@@ -243,7 +244,19 @@ export class ModuleRunner {
     }
   }
 
-  private async cachedModule(
+  private async cachedModule(url: string, importer?: string) {
+    const cacheKey = `${url}~~${importer}`
```

Comment:
Shouldn't the same cached module be returned for different importers if they end up resolving to the same id (that would normally be the case)?

It seems that in `getModuleInformation`, the cachedModule is keyed only by `url`. So we may already be resolving to the same cached module for different importers (even if they resolve to different ids) 🤔

Reply:
Maybe we can cache it if the url starts with `/` I guess

Reply:
I think the cache should be with a resolved id somehow. At `transformRequest()` we have a similar cache by url, and it works because these are already resolved ids during import analysis (turned into URLs). It could still happen there if someone adds an extra query that is later ignored, the cache will not catch it. It works in that case because people don't do that. It seems the relative urls like `./index.ts` are against the root and not the importer (that confused me, and I thought the [urlToIdMap](https://github.com/vitejs/vite/blob/66bfe7b8ddc22cf2950831d179250b2ebd9e352a/packages/vite/src/module-runner/runner.ts#L320) would end broken). But the url there is normalized by resolving against the root, so maybe that should be done here and cache against the normalized url? There could still be an issue if you have a different resolution depending on the importer, but in that case, I think the `urlToIdMap` is going to end up with a wrong entry too because it isn't keying on importer.

Reply:
Need to add more comments, I confused myself with how much iteration this had